lib/liquid/tags/render.rb

```diff
@@ -0,0 +1,79 @@
+module Liquid
+  class Render < Tag
+    Syntax = /(#{QuotedString})#{QuotedFragment}*/o
+
+    attr_reader :template_name_expr, :attributes
+
+    def initialize(tag_name, markup, options)
+      super
+
+      raise SyntaxError.new(options[:locale].t("errors.syntax.render".freeze)) unless markup =~ Syntax
+
+      template_name = $1
+
+      @template_name_expr = Expression.parse(template_name)
+
+      @attributes = {}
+      markup.scan(TagAttributes) do |key, value|
+        @attributes[key] = Expression.parse(value)
+      end
+    end
+
+    def parse(_tokens); end
+
+    def render_to_output_buffer(context, output)
+      template_name = context.evaluate(@template_name_expr)
```

Comment:
Don't we want to disallow that? The template name should always be static.

Reply:
Dynamic template names are prevented at parser time. This simply unescapes the string (eg `"\"foo\""` into `"foo"`).

Reply:
Ah I see. There may be more performant way to extract the name than calling `context.evaluate` then.

Reply:
The unquoting is done by `Expression.parse`, which will return a String instance. `context.evaluate` is a no-op here. I'd probably keep it anyway though, in case this tag were extended. Its total cost is a `respond_to?`.

I'd probably add a comment explaining that the template name parameter is enforced to be a string due to the tag syntax. It also wasn't clear to me at first.